pennylane/operation.py

```diff
@@ -240,6 +241,38 @@ class Operator(abc.ABC):
     """
     do_check_domain = True  #: bool: flag: should we perform a domain check for the parameters?
 
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()
+        copied_op._wires = self.wires
+        copied_op._name = self._name
+
+        if hasattr(self, "_inverse"):
+            copied_op._inverse = self._inverse
+
+        return copied_op
+
+    def __deepcopy__(self, memo):
```

Comment:
This isn't being used internally any more, _but_ I propose we keep it, as it avoids the PyTorch 'cannot copy non-leaf tensor' error if a user attempts to copy PL operations.

Reply:
No, this is more a developer issue that came up during the tape refactor! There were several times that a deepcopy was attempted, e.g., during testing, which had to be re-written for the pytorch interface.